pkg/liqo-controller-manager/resource-request-controller/resource-monitors/external-monitor.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcemonitors
+
+import (
+	"context"
+	"errors"
+	"io"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+)
+
+// ExternalResourceMonitor is an object that keeps track of the cluster's resources.
+type ExternalResourceMonitor struct {
+	ResourceReaderClient
+}
+
+// NewExternalMonitor creates a new ExternalResourceMonitor.
+func NewExternalMonitor(address string) (*ExternalResourceMonitor, error) {
+	klog.Infof("Connecting to %s", address)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
We should forward the context from the caller

```suggestion
func NewExternalMonitor(ctx context.Context, address string) (*ExternalResourceMonitor, error) {
	klog.Infof("Connecting to %s", address)
	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```